src/app/report/edit-display/edit-display.component.ts

```diff
@@ -57,15 +59,17 @@ import { View } from '../../shared/interfaces/view';
     ToggleButtonComponent,
     MatTooltipModule,
     NgClass,
+    EncodingButtonComponent,
   ],
 })
 export class EditDisplayComponent {
+  protected readonly ReportUtil = ReportUtil;
   @Input() id: string = '';
   @Input() containerHeight!: number;
   @Input({ required: true }) currentView!: View;
   @Input() newTab: boolean = true;
   @Output() saveReportEvent: Subject<any> = new Subject<any>();
-  @Output() closeReportEvent: Subject<void> = new Subject<void>();
+  @Output() closeReportEvent: Subject<Report> = new Subject<Report>();
```

Comment:
Why was this changed from void?
The selectedNode is passed by it's parent, so you don't have to pass which report is closed here as the parent already knows.

Reply:
```ts
closeReport(removeReportFromTree: boolean): void {
    this.displayReport = false;
    this.editingRootNode = false;
    this.editingChildNode = false;
    if (removeReportFromTree) {
      this.closeReportEvent.next(this.report);
    }
    this.editor.setNewReport('');
  }
  ```
  This was the code before the change. It gave the this.report which was of type any to the closeReportEvent. I don't know why angular allowed this to be a void, since a variable is being send anyway.
As it stands, the parent does not ask for the current report when removing but does have access to the current report. 

I'll make it so the closeReport method actually calls for the current report instead of asking for one from the event.

Reply:
Added the change.